src/ORM/Rule/IsUnique.php

```diff
@@ -68,14 +60,16 @@ public function __invoke(EntityInterface $entity, array $options): bool
         if (!$entity->extract($this->_fields, true)) {
             return true;
         }
-        $allowMultipleNulls = $this->_options['allowMultipleNulls'];
 
         $alias = $options['repository']->getAlias();
-        $conditions = $this->_alias($alias, $entity->extract($this->_fields), $allowMultipleNulls);
+        $conditions = $this->_alias($alias, $entity->extract($this->_fields));
         if ($entity->isNew() === false) {
             $keys = (array)$options['repository']->getPrimaryKey();
-            $keys = $this->_alias($alias, $entity->extract($keys), $allowMultipleNulls);
-            if (array_filter($keys, 'strlen')) {
+            $keys = $this->_alias($alias, $entity->extract($keys));
+            $filtered = array_filter($keys, function ($v) {
+                return $v !== null;
```

Comment:
Explicit `null` check makes code much easier to understand than the earlier wacky `strlen`.

Reply:
Or do we need to consider empty strings too?

Reply:
NVM, I reverted the change.